ocaml/typing/env.ml

```diff
@@ -3395,12 +3434,19 @@ let lookup_value ~errors ~use ~loc lid env =
   let path, locks, vda =
     lookup_value_lazy ~errors ~use ~loc lid env
   in
-  let vd = Subst.Lazy.force_value_description vda.vda_description in
+  (* First, we apply the modalities to acquire the mode of the value at the
+     definition site. Then, we walk the locks. That means the surrounding
+     closure would be closing over the value instead of the module.
```

Comment:
I think maybe another sentence would make this clearer at first glance? Maybe mention that the locks are what account for closing over things?